src/components/portfolio/portfolio.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * This header file contains implementations of Portfolio and related classes.
+ */
+
+#ifndef PORTFOLIO_H
+#define PORTFOLIO_H
+
+#include "../stock/stock.h"
+#include "../trading_order/trading_order.h"
+
+#include <QVector>
+
+class LoadUp {
+ public:
+  LoadUp();
+  ~LoadUp();
+
+ private:
+  qint64 time_stamp;
+  qreal quantity;
+};
+
+class Portfolio {
+ public:
+  Portfolio();
+
+  qreal valuation();
+
+  void addStockToWatchList(Stock st);
+  void removeStockFromWatchList(Stock st);
+
+  void addTradingOrder(TradingOrder trading_order);
+
+  void addLoadUp(LoadUp load_up);
+
+ private:
+  qreal current_money;
+
+  QVector<Stock> stock_watch_list;
```

Comment:
I am not sure about this but would it be better to store a vector of pointers instead of objects themselves? Since in the other functions you would like to compare objects before adding it?

Reply:
Good idea